packages/components/src/Input/Input/Input.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { InputHTMLAttributes } from "react"
+import classnames from "classnames"
+import { OverrideClassName } from "~types/OverrideClassName"
+import { InputStatus, InputTypes } from "./types"
+import styles from "./Input.module.scss"
+
+export type InputType = (typeof InputTypes)[number]
+export type InputStatusType = (typeof InputStatus)[number]
+
+export type InputProps = {
+  inputRef?: React.RefObject<HTMLInputElement>
+  status?: InputStatusType
+  startIconAdornment?: React.ReactNode
+  endIconAdornment?: React.ReactNode
+  reversed?: boolean
+  type?: InputType
+} & OverrideClassName<InputHTMLAttributes<HTMLInputElement>>
+
+export const Input = ({
+  inputRef,
+  status = "default",
+  startIconAdornment,
+  endIconAdornment,
+  reversed = false,
+  type = "text",
+  value,
+  defaultValue,
+  classNameOverride,
+  disabled,
+  ...restProps
+}: InputProps): JSX.Element => (
+  <div
+    className={classnames(
+      styles.wrapper,
+      reversed && styles.withReversed,
+      disabled && styles.withDisabled,
+      startIconAdornment && styles.withStartIconAdornment,
+      endIconAdornment && styles.withEndIconAdornment,
+      status != "default" && styles.hasStatus
+    )}
+  >
+    {startIconAdornment && (
+      <div className={styles.startIconAdornment}>{startIconAdornment}</div>
+    )}
+
+    <input
+      ref={inputRef}
+      type={type}
+      value={value}
+      defaultValue={defaultValue}
+      disabled={disabled}
+      className={classnames(
+        styles.input,
+        styles[status],
+        classNameOverride,
+        reversed ? styles.reversed : styles.default,
+        disabled && styles.disabled
+      )}
+      {...restProps}
+    />
+
+    {/* Inputs aren't able to have pseudo elements like ::after on them,
+          so we have to create an element ourselves for the focus ring */}
+    <div className={styles.focusRing} />
```

Comment:
Noticed a spacing issue on the focus ring for the input, could be that the offset hasn't properly calculated itself?

![Screenshot 2023-10-02 at 11 17 19 am](https://github.com/cultureamp/kaizen-design-system/assets/36558508/c4137333-0fc9-4d6c-8bc0-5e3e47948fac)

Worth noting also that this empty `div`  feels like a bit of an older pattern since `focus-within` wasn't widely supported a couple of years back - If we move this to a pseudo after element on the `wrapper` this may clean up some of the sibling selectors in the stylesheet and make it cleaner the offset?

Reply:
Let's log that as a ticket to solve separately. The issue here is resolved by not using `grid` or `flex` for alignment.

Reply:
https://cultureamp.atlassian.net/browse/KDS-1822